src/main/java/org/apache/cassandra/sidecar/cdc/CDCRawDirectoryMonitor.java

```diff
@@ -0,0 +1,90 @@
+package org.apache.cassandra.sidecar.cdc;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Timer;
+import java.util.TimerTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.sidecar.metrics.cdc.CDCReaderMonitor;
+
+/**
+ * Monitors the cdc_raw directory, cleanup unused commit logs and report metrics
+ * */
+@Singleton
+public class CDCRawDirectoryMonitor extends TimerTask
+{
+
+    private final Timer timer;
+    private final CDCReaderMonitor monitor;
+    private volatile boolean running;
+    private static final Logger logger = LoggerFactory.getLogger(CDCRawDirectoryMonitor.class);
+
+    @Inject
+    CDCRawDirectoryMonitor(CDCReaderMonitor monitor)
+    {
+        this.timer = new Timer();
+        this.monitor = monitor;
+        this.running = false;
+    }
+
+    /**
+     * Starts the background thread to monitor the cdc_raw dir.
+     * */
+    public void startMonitoring()
+    {
+        this.running = true;
+        timer.schedule(this, 0, DatabaseDescriptor.getCDCDiskCheckInterval());
+    }
+
+    @Override
+    public void run()
+    {
+        if (!this.running)
+        {
+            return;
+        }
+        // TODO : Don't be someone who just complains, do some useful work, clean files older than
+        //  the last persisted bookmark.
+        this.monitor.reportCdcRawDirectorySizeInBytes(getCdcRawDirectorySize());
```

Comment:
We don't got cdc data clean ? I saw your TODO, and Why we do not clean the cdc log that is useless after we emit the change of the log ?

Reply:
We do delete commit logs after reading them : https://github.com/apache/cassandra-sidecar/pull/18/files#diff-32afd2c4bf3fe7d4c3268bec08a902cf36fb92b079d9c99fe47bafe0b073d5ceR289. However, if the output is blocked/slow for some reason (e.g. a Kafka node is down), then commit logs can pile up and eventually halt Cassandra writes. This is the default behavior, but we don't want that.